pages/DynamicPOC/index.tsx

```diff
@@ -124,51 +124,37 @@ const DynamicPOC: React.FC<{ questions: Question[] }> = ({ questions }) => {
    * Modifies question history and routes form depending on whether answer has been changed
    */
   function _handleQuestionExists() {
-    const nextQuestion = getNextQuestion(
-      questions[currentQuestion.id].answers[parseInt(currentAnswer.answerId)]
-    )
+    const nextQuestion = getNextQuestion(currentQuestion, currentAnswer)
+    const nextAnswer = formValues.formAnswers[currentQuestion.id]
+    // if the exisitng question is a radio, and they're not the same answer
     if (
-      formValues.formAnswers[currentQuestion.id]['answerId'] !==
-      currentAnswer['answerId']
+      nextAnswer.type === QuestionType.RADIO &&
+      currentAnswer.type === QuestionType.RADIO &&
+      nextAnswer.answerId !== currentAnswer.answerId
     ) {
       for (let i = currentIndex + 1; i < questionHistory.length; i++) {
         delete formValues.formAnswers[questionHistory[i].id]
       }
       const questionSlice = questionHistory.slice(0, currentIndex + 1)
-      const nextQuestion = getNextQuestion(
-        questions[currentQuestion.id].answers[parseInt(currentAnswer.answerId)]
-      )
       setQuestionHistory([...questionSlice, nextQuestion])
-      formValues.formAnswers[currentQuestion.id] = currentAnswer
-      setCurrentQuestion(nextQuestion)
-      if (formValues.formAnswers[nextQuestion.id]) {
-        setCurrentAnswer(formValues.formAnswers[nextQuestion.id])
-      }
-    } else {
-      const nextQuestion = getNextQuestion(
-        questions[currentQuestion.id].answers[parseInt(currentAnswer.answerId)]
-      )
-      setCurrentQuestion(nextQuestion)
-      if (formValues.formAnswers[nextQuestion.id]) {
-        setCurrentAnswer(formValues.formAnswers[nextQuestion.id])
-      }
     }
+    _handleQuestionChange(nextQuestion)
   }
 
   function _handleBack() {
+    // TODO: in redesign, if we disable the back button on the first question, we can get rid of this check
```

Comment:
This would just be a conditional in our form, where we check if the question ID is 0 and only say yeah you can go back if it isn't id 0